nexus/deployment/src/planner.rs

```diff
@@ -0,0 +1,227 @@
+        Planner { log, policy, blueprint }
+    }
+
+    pub fn plan(mut self) -> Result<Blueprint, Error> {
+        self.do_plan()?;
+        Ok(self.blueprint.build())
+    }
+
+    fn do_plan(&mut self) -> Result<(), Error> {
+        // The only thing this planner currently knows how to do is add services
+        // to a sled that's missing them.  So let's see if we're in that case.
+
+        // Internal DNS is a prerequisite for bringing up all other zones.  At
+        // this point, we assume that internal DNS (as a service) is already
+        // functioning.  At some point, this function will have to grow the
+        // ability to determine whether more internal DNS zones need to be
+        // added and where they should go.  And the blueprint builder will need
+        // to grow the ability to provision one.
+
+        for (sled_id, sled_info) in &self.policy.sleds {
+            // Check for an NTP zone.  Every sled should have one.  If it's not
+            // there, all we can do is provision that one zone.  We have to wait
+            // for that to succeed and synchronize the clock before we can
+            // provision anything else.
+            if self.blueprint.sled_ensure_zone_internal_ntp(*sled_id)? {
+                info!(
+                    &self.log,
+                    "found sled missing NTP zone (will add one)";
+                    "sled_id" => ?sled_id
+                );
+                self.blueprint
+                    .comment(&format!("sled {}: add NTP zone", sled_id));
+                // Don't make any other changes to this sled.  However, this
+                // change is compatible with any other changes to other sleds,
+                // so we can "continue" here rather than "break".
+                continue;
+            }
+
+            // Every zpool on the sled should have a Crucible zone on it.
+            let mut ncrucibles_added = 0;
+            for zpool_name in &sled_info.zpools {
+                if self
+                    .blueprint
+                    .sled_ensure_zone_crucible(*sled_id, zpool_name.clone())?
+                {
+                    info!(
+                        &self.log,
+                        "found sled zpool missing Crucible zone (will add one)";
+                        "sled_id" => ?sled_id,
+                        "zpool_name" => ?zpool_name,
+                    );
+                    ncrucibles_added += 1;
+                }
+            }
+
+            if ncrucibles_added > 0 {
+                // Don't make any other changes to this sled.  However, this
+                // change is compatible with any other changes to other sleds,
+                // so we can "continue" here rather than "break".
+                // (Yes, it's currently the last thing in the loop, but being
+                // explicit here means we won't forget to do this when more code
+                // is added below.)
```

Comment:
This implies we should have an explicit `continue` after the `comment` - can we add one?

Reply:
Added in c58c7edae525570e703cb113075211ccd9903c6c.  (I think what happened here is that I talked myself out of needing the `continue` because really you should be able to provision any number of zones at once, once NTP is there (and presuming internal DNS is working).  But rather than overthink it, I'm going to leave the comment and `continue` there and let us revisit when we augment this.)